lib/webhooks.js

```diff
@@ -55,31 +61,53 @@ function startFromCommit(project, payload, send) {
 // returns : {trigger, branch, deploy}
 function pushJob(payload) {
   let branchname;
-  const commit = payload.commits[payload.commits.length - 1];
-
-  if(!commit || !commit.author) {
-    debug('Unable to determine author from provided commit (may it\'s a merge commit?). Skipping');
-    debug(`Commit data extracted from payload:${util.inspect(commit)}`);
-    return null;
+  let commit;
+  if (payload.commits) {
+    commit = payload.commits[payload.commits.length - 1];
+
+    if(!commit || !commit.author) {
+      debug('Unable to determine author from provided commit (may it\'s a merge commit?). Skipping');
+      debug(`Commit data extracted from payload:${util.inspect(commit)}`);
+      return null;
+    }
+  } else if (payload.user_name) {
```

Comment:
Why does having a `user_name` signify that this is a tag? Please leave a comment explaining for future dev work. Thanks!

Reply:
Especially since elsewhere you use `payload.ref.indexOf('refs/tags/')`